duckduckgo/Android
# Re-structure VPN tracker db table

Task/Issue URL: https://app.asana.com/0/488551667048375/1203195823975934/f

### Description
Change the `vpn_tracker` schema so that we don't need one entry per tracking attempt.


This should:
* improve DB access performance
* fix some bugs when we have more than 10,000 tracking attempts

### Steps to test this PR
_Test DB migration_
- [x] enable AppTP
- [x] open several apps that track
- [x] (alternatively you can follow the instructions down below to create tracking attempts)
- [x] In the `vpn_tracker` db table in `vpn.db`, modify the day part of the `timestamp` in some of the tracking attempts (eg. 1,2,3,4 days ago) so that the DB is more representative
- [x] (alternative/in addition you can change the termux script to go from `1..1000` to create more than 10,000 tracking attempts)
- [x] update from this branch
- [x] launch app and enable AppTP
- [x] ensure app doesn't crash (migration is done correctly)
- [x] open AppTP activity screen and ensure all counters are correct, ie. past 7 days general counter, recent activity counters
- [x] open AppTP "view all recent activity" and ensure all counters are correct–for this you may need to have tracking attempts from more than 5 apps and/or different days
- [x] open some apps that generate new tracking attempts or do so with Termux
- [x] verify AppTP accounts tracking attempts correctly


_How to create tracking attempts_
* install Termux
* execute the script below
```bash
for i in {1..10}
do
  curl -v https://api2.branch.io &
  sleep 0.5
done
```
